content/configuration/configuration-types.md

```diff
@@ -11,21 +11,38 @@ contributors:
 Besides exporting a single config object, there are a few more ways that cover other needs as well.
 
 
-## Exporting a function to use `--env`
+## Exporting a function to use `--env` and `argv`
 
 Eventually you will find the need to disambiguate in your `webpack.config.js` between [development](/guides/development) and [production builds](/guides/production). You have (at least) two options:
 
-Instead of exporting a configuration object, you may return a function which accepts an environment as argument. When running webpack, you may specify build environment keys via `--env`, such as `--env.production` or `--env.platform=web`.
+Instead of exporting a configuration object, you may return a function which accepts an environment as its first parameter. It may also take an options map as the second parameter.
+
+When running webpack, you may specify build environment keys via `--env`, such as `--env.production` or `--env.platform=web`.
+
+| Invocation                   | Resulting env             |
+|------------------------------|---------------------------|
+| webpack --env prod           | "prod"                    |
+| webpack --env.prod           | { prod: true }            |
+| webpack --env.prod=1         | { prod: 1 }               |
+| webpack --env.prod=foo       | { prod: "foo" }           |
+| webpack --env.prod --env.bar | { prod: true, bar: true } |
+| webpack --env.prod --env bar | [{ prod: true }, "bar"]   |
```

Comment:
The majority of this should be moved to the CLI section of the documentation, probably in [this section](https://webpack.js.org/api/cli/#common-options) as that's the only place I'm seeing `--env` mentioned. Then, from this page, we should link to there (feel free to create a dedicated section for `--env` with a header so you can link directly to it's hash).